### PR TITLE
GH-47 Add EternalCombat API to docs

### PR DESCRIPTION
**Pull request by CitralFlo**

### CODE REVIEW

**Review by vLuckyyy — CHANGES_REQUESTED** [review]
In instruction to add dependency, please add info about depend in `plugin.yml`

E.g.
```yaml
depend: [EternalCombat]
```

And please improve example code, this code not works. I suggest to add in examples send mesaage "Hello <player>" in combat. And usage how to check if player is in combat for example in PlayerMoveEvent for example to prevent player move to plot region.

**Comment by CitralFlo** [comment]
> In instruction to add dependency, please add info about depend in `plugin.yml`
> 
> E.g.
> 
> ```yaml
> depend: [EternalCombat]
> ```
> 
> And please improve example code, this code not works. I suggest to add in examples send mesaage "Hello " in combat. And usage how to check if player is in combat for example in PlayerMoveEvent for example to prevent player move to plot region.

roger that!

**Review by Rollczi — CHANGES_REQUESTED** [review]
See dm discord

File: docs/eternalcombat/using-API.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+You can access both event types by using EventListeners.
+
+```java
+@EventHandler
+public void onTagEvent(FightTagEvent event) {
+    
+    // simple check if players UUID is equal to custom UUID
+    uniqueID = "19e84b9c4ab34690b8b012bcfa5e7649";
+    if (player.getUniqueId()equals(uniqueID)){
```
Comment (Hyd3r1) [inline docs/eternalcombat/using-API.md:90]: .equals

File: docs/eternalcombat/using-API.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        // if player UUID is equal to custom UUID, then cancel event
+        event.setCancelled(true);
```
Comment (Hyd3r1) [inline docs/eternalcombat/using-API.md:92]: tab

**Review by Hyd3r1 — CHANGES_REQUESTED** [review]

File: docs/eternalcombat/using-API.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+```java
+@EventHandler
+void onTag(FightUntagEvent event) {
+    CauseOfUnTag cause = event.getCause();
+    
+    if (cause == CauseOfUnTag.DEATH) {
+        server.getPlayer(event.getPlayer()).sendMessage("You died!");
+    }
+}
+```
```
Comment (vLuckyyy) [inline docs/eternalcombat/using-API.md:107]: I would like you to keep the standard style, not everyone may guess that you need to add a Server to the constructor.
```suggestion
```java
    @EventHandler
    public void onTag(FightUntagEvent event) {
        CauseOfUnTag cause = event.getCause();

        if (cause == CauseOfUnTag.DEATH) {
            Bukkit.getPlayer(event.getPlayer()).sendMessage("You died!");
        }
    }
```
```

File: docs/eternalcombat/using-API.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+private EternalCombatApi eternalCombatApi;
+private EternalCombatProvider eternalCombatProvider;
+
+this.eternalCombatApi = eternalCombatProvider.provide();
+FightManager fightManager = this.eternalCombatApi.getFightManager();
+
+if (fightManager.isInCombat(uuID)) {
+    server.getPlayer(uuID).sendMessage("You are in combat!");
+}
+```
```
Comment (vLuckyyy) [inline docs/eternalcombat/using-API.md:88]: I think the example should look at least like this:

```java
public class YourMainClass {

    private EternalCombatApi eternalCombatApi;
    private EternalCombatProvider eternalCombatProvider;

    public onEnable() {
        this.eternalCombatApi = eternalCombatProvider.provide();
        YourApiUsage yourApiUsage = new YourApiUsage(this.eternalCombatApi.getFightManager());
    }


}

public class YourApiUsage {

    private final FightManager fightManager;

    public YourApiUsage(FightManager fightManager) {
        this.fightManager = fightManager;
    }

    public void exampleMethod(UUID playerUUID) {
        if (this.fightManager.isInCombat(playerUUID)) {
            Bukkit.getPlayer(playerUUID).sendMessage("You are in combat!");
        }
    }

}
```

**Review by vLuckyyy — CHANGES_REQUESTED** [review]

File: docs/eternalcombat/using-API.md (hunk 3 above)
Comment (CitralFlo) [reply docs/eternalcombat/using-API.md:107]: okay i understand, i wasn't aware of that

**Review by Rollczi — APPROVED** [review]
We talked on a voice channel, so I have no objections

**Review by vLuckyyy — APPROVED** [review]
obraz

**Review by Hyd3r1 — APPROVED** [review]

**Pull request merged**